__tests__/app/(dashboard)/dashboard/chat/[chatId]/helpers.test.ts

```diff
@@ -0,0 +1,86 @@
+import {Helpers} from "@/app/(dashboard)/dashboard/chat/[chatId]/helpers";
+import {notFound} from "next/navigation";
+import fetchRedis from "@/helpers/redis";
+import {messageArraySchema} from "@/lib/validations/messages"
+
+jest.mock("@/helpers/redis", () => ({
+    __esModule: true,
+    default: jest.fn(),
+}));
+
+jest.mock("next/navigation", () => ({
+    notFound: jest.fn(),
+}));
+
+describe('getChatMessage Tests', () => {
+    let helpers: Helpers;
+    beforeEach(() => {
+        helpers = new Helpers();
+        jest.spyOn(helpers, 'fetchChatById').mockResolvedValue(["{\"id\": \"1701\", \"senderId\": \"bob\", \"text\":\"Hello world. My name's Gypsy, what's yours?\", \"timestamp\":1729437427}"]);
+
+    })
+
+    afterEach(() => {
+        jest.resetAllMocks();
+    });
+
+    test('if fetchChatById throws, notFound is called', async () => {
+        jest.spyOn(helpers, 'fetchChatById').mockRejectedValue('Bad')
+
+        await helpers.getChatMessage("chat:userOne-userTwo:message");
+
+        expect(notFound).toHaveBeenCalled();
+    });
+
+    test("if fetchRedis doesn't throw, notFound isn't called", async () => {
+        await helpers.getChatMessage("chat:userOne-userTwo:message");
+
+        expect(notFound).not.toHaveBeenCalled();
+    });
+
+    test('If getChatMessage is called, then messageArraySchema.parse is called with the output of fetchChat',async ()=>{
```

Comment:
if-then statements are clear, but verbose

Let's consider breaking stuff up into describes, and each test description is a supporting point to the purpose of describe